sdk/storage/azure-storage-blob-nio/src/test/java/com/azure/storage/blob/nio/AzureFileSystemProviderSpec.groovy

```diff
@@ -303,15 +306,509 @@ class AzureFileSystemProviderSpec extends APISpec {
         props.getCacheControl() == "myControl"
     }
 
+    @Unroll
+    def "FileSystemProvider copy source"() {
+        setup:
+        def fs = createFS(config)
+
+        // Generate resource names.
+        // Don't use default directory to ensure we honor the root.
+        def rootName = fs.getRootDirectories().last().toString()
+        def containerName = rootToContainer(rootName)
+        def sourceName = generateBlobName()
+        def sourcePath = fs.getPath(rootName, sourceName)
+        def destName = generateBlobName()
+        def destPath = fs.getPath(rootName, destName)
+
+        // Generate clients to resources.
+        def containerClient = primaryBlobServiceClient
+            .getBlobContainerClient(containerName)
+        def sourceClient = containerClient.getBlobClient(sourceName)
+        def destinationClient = containerClient.getBlobClient(destName)
+        def sourceChildClient = null
+        def destChildClient = null
```

Comment:
Can you throw this in the file's setup method? It looks mostly the same throughout.

Reply:
Or if they differ enough for this to not be an option, it looks like you always use these variables. Have them defined on the class and have a "default populate" method that you can ignore for the times where you really do need different values. That will also help identify when a test is doing something different where that setup matters.

Reply:
This will come in my PR for delete, but I did add this.